### PR TITLE
Adds the ability to define a timeout on windows_feature

### LINKED ISSUE #370: Allow windows_feature to have definable timeout
### Cookbook version

1.41.0
### Chef-client version

12.2.1 (Latest version available for Windows on AWS OpsWorks)
### Platform Details

Server 2012 R2
### Scenario:

We specify an alternate source for WIM files that is not as close to the server as desired, so when installing a feature that contains many (like Web-Net-Ext) the installation takes longer than what appears to be a default 10 minute timeout. I would like to be able to define a timeout value.
### Steps to Reproduce:

You could define a source that's on a slow connection. We set the source via Registry with server creation.
### Expected Result:

Feature installs successfully after defined timeout
### Actual Result:

https://gist.github.com/Jakauppila/09f1dd243aac00ba86d7e33a8c4bfbde

### PR DESCRIPTION
**Pull request by jakauppila**
### Description

Adds the ability to define a timeout on windows_feature by adding a property to be passed to shell_out.
Removes the `source` property from the servermanagercmd provider as it does not support it.
Updates README to reflect `source` available for the Powershell provider as of #434 

### Issues Resolved

Fixes #370 

### Check List

- [ ] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

**Comment by jakauppila** [comment]
@tas50 Looks like Appveyor builds have been failing for a while, possibly due to a change to https://omnitruck.chef.io/install.ps1 ?

**Comment by tas50** [comment]
@jakauppila If you rebase this should start passing

**Comment by jakauppila** [comment]
@tas50 I'll give it a shot this evening, but based on some testing I was doing last night, I think it's still getting hung up on something else (different problem than what I was seeing here).

If you take a look at the Appveyor build off master from yesterday it looks like it's hitting some WinRM bug.

**Comment by tas50** [comment]
@jakauppila Can you rebase this?

**Comment by jakauppila** [comment]
@tas50 This has been rebased

File: resources/feature.rb [hunk 1]
```diff
@@ -63,20 +64,22 @@ def run_default_provider(desired_action)
         feature_name new_resource.feature_name
         source new_resource.source if new_resource.source
         all new_resource.all
+        timout new_resource.timeout
       end
     when :windows_feature_servermanagercmd
       windows_feature_servermanagercmd new_resource.name do
         action desired_action
         feature_name new_resource.feature_name
-        source new_resource.source if new_resource.source
```
Comment (tas50) [inline resources/feature.rb]: ?

File: resources/feature_servermanagercmd.rb [hunk 2]
```diff
@@ -19,24 +19,24 @@
 #
 
 property :feature_name, [Array, String], name_attribute: true
-property :source, String
```
Comment (tas50) [inline resources/feature_servermanagercmd.rb]: ?

File: resources/feature.rb (hunk 1 above)
Comment (jakauppila) [reply resources/feature.rb]: Servermanagercmd.exe doesn't actually support defining an alternate source like DISM and Powershell do

Comment (tas50) [reply resources/feature.rb]: I should read the PR before commenting. Oops

**Pull request merged**